src/roles/vsc-deploy/tasks/setup_vsc_config.yml

```diff
@@ -85,7 +89,38 @@
       {{ proxy_conf | default('') }} {{ tmp_dir_name }}/{{ config_file_name }}
       {{ vsc_username|default(vsc_default_username) }}@{{ '[' }}{{ mgmt_ip }}{{ ']' }}:
     no_log: "{{ lookup('env', 'METROAE_NO_LOG') or 'true' }}"
-    when: mgmt_ip | ipv6
+    when:
+      - mgmt_ip | ipv6
+      - not override_vsc_config
+
+  - block:
+    - name: Copy user provided config file for VSC(IPv4)
+      connection: local
+      shell: >-
+        sshpass -p'{{ vsc_password | default(vsc_default_password) }}' scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null
+        {{ proxy_conf | default('') }} {{ item }} {{ vsc_username | default(vsc_default_username) }}@{{ mgmt_ip }}:
+      loop: "{{ vsc_config_file_paths }}"
+      when: not mgmt_ip | ipv6
+
+    - name: Copy user provided config file for VSC(IPv6)
+      connection: local
+      shell: >-
+        sshpass -p'{{ vsc_password | default(vsc_default_password) }}' scp -6 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null
+        {{ proxy_conf | default('') }} {{ item }}
+        {{ vsc_username|default(vsc_default_username) }}@{{ '[' }}{{ mgmt_ip }}{{ ']' }}:
+      loop: "{{ vsc_config_file_paths }}"
+      when: mgmt_ip | ipv6
+
+  - name: Save user provided config file
+    sros_command:
+      commands:
+        - exec cf1:\{{ item | basename }}
+        - admin save
+      provider: "{{ vsc_creds }}"
+    loop: "{{ vsc_config_file_paths }}"
+    delegate_to: localhost
+
+    when: override_vsc_config
```

Comment:
Is this "when" indented too much?  Looks like it matches the task and not the block.

Reply:
Fixed it to match the task. The task was accidentally one ident out so the when was matching both the task and the block